x/auth/types/genesis.go

```diff
@@ -89,8 +108,47 @@ func (GenesisAccountIterator) IterateGenesisAccounts(
 	cdc codec.Marshaler, appGenesis map[string]json.RawMessage, cb func(AccountI) (stop bool),
 ) {
 	for _, genAcc := range GetGenesisStateFromAppState(cdc, appGenesis).Accounts {
-		if cb(genAcc) {
+		acc, ok := genAcc.GetCachedValue().(AccountI)
+		if !ok {
+			panic("expected account")
+		}
+		if cb(acc) {
 			break
 		}
 	}
 }
+
+// ConvertAccounts converts GenesisAccounts to Any slice
+func ConvertAccounts(accounts GenesisAccounts) ([]*types.Any, error) {
+	accountsAny := make([]*types.Any, len(accounts))
+	for i, acc := range accounts {
+		msg, ok := acc.(proto.Message)
+		if !ok {
+			return nil, fmt.Errorf("cannot proto marshal %T", acc)
+		}
+		any, err := types.NewAnyWithValue(msg)
+		if err != nil {
+			return nil, err
+		}
+		accountsAny[i] = any
+	}
+
+	return accountsAny, nil
+}
+
+// ConvertAccountsAny converts Any slice to GenesisAccounts
+func ConvertAccountsAny(accountsAny []*types.Any) (GenesisAccounts, error) {
```

Comment:
this needs a test

Reply:
check out https://github.com/cosmos/cosmos-sdk/blob/marie/5917-auth-genesis-proto/x/auth/types/genesis_test.go#L83-L134